packages/widgets/src/dns-hole/summary/component.tsx

```diff
@@ -0,0 +1,173 @@
+"use client";
+
+import type { BoxProps } from "@mantine/core";
+import { Box, Card, Center, Flex, Text } from "@mantine/core";
+import { useElementSize } from "@mantine/hooks";
+import {
+  IconBarrierBlock,
+  IconPercentage,
+  IconSearch,
+  IconWorldWww,
+} from "@tabler/icons-react";
+
+import type { RouterOutputs } from "@homarr/api";
+import { clientApi } from "@homarr/api/client";
+import { formatNumber } from "@homarr/common";
+import type {
+  stringOrTranslation,
+  TranslationFunction,
+} from "@homarr/translation";
+import { translateIfNecessary } from "@homarr/translation";
+import { useI18n } from "@homarr/translation/client";
+import type { TablerIcon } from "@homarr/ui";
+
+import type { WidgetComponentProps, WidgetProps } from "../../definition";
+import { NoIntegrationSelectedError } from "../../errors";
+
+export default function DnsHoleSummaryWidget({
+  options,
+  integrationIds,
+}: WidgetComponentProps<"dnsHoleSummary">) {
+  const integrationId = integrationIds.at(0);
+
+  if (!integrationId) {
+    throw new NoIntegrationSelectedError();
+  }
+
+  const [data] = clientApi.widget.dnsHole.summary.useSuspenseQuery(
+    {
+      integrationId,
+    },
+    {
+      refetchOnMount: false,
+      retry: false,
+    },
+  );
+
+  return (
+    <Box h="100%" {...boxPropsByLayout(options.layout)}>
+      {stats.map((item, index) => (
+        <StatCard
+          key={index}
+          item={item}
+          usePiHoleColors={options.usePiHoleColors}
+          data={data}
+        />
+      ))}
+    </Box>
+  );
+}
+
+const stats = [
+  {
+    icon: IconBarrierBlock,
+    value: ({ adsBlockedToday }) => formatNumber(adsBlockedToday, 2),
+    label: (t) => t("widget.dnsHoleSummary.data.adsBlockedToday"),
+    color: "rgba(240, 82, 60, 0.4)",
```

Comment:
Use variable for all colors so that they are named (RED, …)

Reply:
I've used comments as they provide more value and are easier to understand